src/coreclr/vm/eedbginterfaceimpl.cpp

```diff
@@ -631,7 +631,32 @@ PCODE EEDbgInterfaceImpl::GetFunctionAddress(MethodDesc *pFD)
     }
     CONTRACTL_END;
 
-    return pFD->GetNativeCode();
+    //return pFD->GetNativeCode();
+        WRAPPER_NO_CONTRACT;
+    SUPPORTS_DAC;
+
+    PCODE pDefaultCode = pFD->GetNativeCode();
```

Comment:
I'd revert GetFunctionAddress() to it's previous behavior. My opinion is that this choice to look at the default version and then fallback to the active version should be treated as a specialized way of looking up code and we use it only in narrow scenarios where we know this is the behavior that we want. My guess is that most of the time (maybe even all the time?) this search algorithm isn't the ideal behavior. Probably the places that were calling HasNativeCodeRejitAware() and GetNativeCodeRejitAware() really wanted to know if there is _any_ version of the code that exists, and it happens that checking default and active is a pretty good approximation of 'find any jitted version' most of the time.

Reply:
This means reintroducing Has/GetNativeCodeRejitAware, right?

Reply:
If you still need to call them somewhere, then yes. The only calls I noticed were in the DAC and for those two spots I suspect you'd be better off enumerating all versions rather than only enumerating default+active. Not sure if there were other calls to this that I missed.